### PR TITLE
Upgraded the league/flysystem dependency from a 1.x version to a 2.x version

### PR DESCRIPTION
**Pull request by Pointotech**
**Bug fixed by this pull request:**
`laravel-framework` is currently incompatible with other libraries which use `Flysystem 2.x`, because Laravel uses `Flysystem 1.x`.

**Replaced league/flysystem-cached-adapter with lustmored/flysystem-v2-simple-cache-adapter:**
I've replaced `league/flysystem-cached-adapter` with `lustmored/flysystem-v2-simple-cache-adapter`, because `league/flysystem-cached-adapter` does not support `Flysystem 2.x` and it appears to be abandoned:

- The last commit was over a year ago.
- There are several open issues (such as https://github.com/thephpleague/flysystem-cached-adapter/issues/41 ) about upgrading to `Flysystem 2.x`, which would presumably be the main focus of this library, if it was maintained.
- The relevant open issues have received very little engagement since their creation, no engagement at all in months, and no engagement at all from maintainers.
- `lustmored/flysystem-v2-simple-cache-adapter` was suggested as a replacement in the above issue, which is why I replaced the unmaintained library with it.

**Why I think this pull request is pointed at the correct branch:**
- The target branch corresponds to a currently supported version according to https://laravel.com/docs/8.x/releases#support-policy .
- The target branch is the `latest stable branch` (as used in https://laravel.com/docs/contributions#which-branch , if I understand the term correctly).

**Caveat:**
This does not break anything that I know of, but I do not actively use the `Flysystem cached adapter`, nor do I know how to use it. I am simply trying to combine the latest version of `laravel-framework` with `Flysystem 2.x` without breaking my build.

### CODE REVIEW

**Comment by morloderex** [comment]
@Pointotech I am sorry to break the news for you but flysystem has already been upgraded for version 2 in the upcoming major release.

**Comment by GrahamCampbell** [comment]
Yes, indeed. Also, the v2 API is totally different, so this PR doesn't work. I assume you didn't test it?

**Comment by Pointotech** [comment]
@morloderex Is there a branch where I can find this code, in a version compatible with the latest PHP? I tried using the `master` branch from this repo, instead of the `8.x` branch of my own repo, and it gave me this conflict:

> Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.0.10) does not satisfy that requirement.
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.0.10) does not satisfy that requirement.
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.0.10) does not satisfy that requirement.
    - Root composer.json requires laravel/framework dev-master -> satisfiable by laravel/framework[dev-master].
    - barryvdh/laravel-debugbar v3.6.2 requires illuminate/support ^6|^7|^8 -> satisfiable by illuminate/support[v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev].
    - Only one of these can be installed: illuminate/support[dev-master, v5.6.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev], laravel/framework[dev-master]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - Root composer.json requires barryvdh/laravel-debugbar ^3.6.2 -> satisfiable by barryvdh/laravel-debugbar[v3.6.2].

Basically, it looks like the version of `illuminate/support` used on the `master` branch is not compatible with PHP 8.0. Is that a known issue with the `laravel:master` branch? It's not a problem on the `laravel:8.x` branch.

@GrahamCampbell As I said, I am not actively using the Flysystem cached adapter. I do not know how to test it.

There are automated tests that ran once I created a pull request, and some of those failed. I can take a look at those stack traces.

**Comment by taylorotwell** [comment]
Yeah I think this is a breaking change but will let @driesvints chime in. We have already updated to Flysystem 2.0 on the upcoming 9.x release slated for January.

**Comment by driesvints** [comment]
We cannot support Flysystem v2 on 8.x as there are breaking changes. Atm the cached adapter is abandoned by Flysystem and we have no plans to directly integrate it into the framework atm.
